tests/templates/test_subroutines/test_select.py

```diff
@@ -23,82 +23,134 @@
 
 import pennylane as qml
 from pennylane import numpy as pnp
-from pennylane.templates.subroutines.select import _select_decomp_partial_unary
+from pennylane.templates.subroutines.select import (
+    _partial_select,
+    _select_decomp_partial_unary,
+    _select_resources_partial_unary,
+)
 
 
 @pytest.mark.parametrize("num_ops", [3, 10, 15, 16])
-def test_standard_checks(num_ops):
+@pytest.mark.parametrize("partial", [True, False])
+@pytest.mark.parametrize("work_wires", [None, [5, 6, 7]])
+def test_standard_checks(num_ops, partial, work_wires):
     """Run standard validity tests."""
+    if work_wires is None:
+        pytest.xfail()
```

Comment:
we dont have a decomposition of Select when there are no aux wires?

Reply:
Ah, no, it's just that the new decomposition rule check fails for unary iteration if we don't have them, and I translated that wrongly into an `xfail`. Very good catch, changed that to be reflected in the `skip_new_decomp` condition instead!